Automattic/studio
# fix(sync): ocasionally marked sites as deleted

## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9978

## Proposed Changes
We have an issue - we provide `connectedSiteIds` inside `useFetchWpComSites`, but it's not all connected sites, it's just connected sites to the SELECTED site in Studio UI. But, we are updating `syncSupport` to all sites which we have, that's why some sites (connected to all local sites, except selected) is marked with wrong syncSupport value.
With this PR we updating list of sited with the help of list of [all connected sites](https://github.com/Automattic/studio/pull/749/files#diff-11163dcf41700f6b8b8bc19f4a9a3a1f4abc91c7d33c5ff9d962d51fe053bbc1R240), so we have always correct information in appdata-v1.json

## Testing Instructions
1. Create two local sites in Studio (A, B)
2. Create 2 sites in wordpress.com (C, D)
3. Conencted C to A, and D to B.
4. Open Studio and select some site
5. Open `Sync` tab
6. Select another local site in Studio
7. Assert that you don't see flash of content (Actually previously we saw the site marked as Deleted, literally for a few milliseconds)
8. Try to test syncing with Wordpress.com - change title, delete Business subscription, delete site and assert that all changes reflects to Studio.
